src/parser/mod.rs

```diff
@@ -965,6 +969,178 @@ impl<'a> Parser<'a> {
+                    || matches!(
+                        self.peek_nth_token(1).token,
+                        Token::Word(Word {
+                            keyword: Keyword::SELECT | Keyword::WITH,
+                            ..
+                        })
+                    ) =>
+            {
+                self.parse_exists_expr(false)
+            }
+            Keyword::EXTRACT => self.parse_extract_expr(),
+            Keyword::CEIL => self.parse_ceil_floor_expr(true),
+            Keyword::FLOOR => self.parse_ceil_floor_expr(false),
+            Keyword::POSITION if self.peek_token().token == Token::LParen => {
+                self.parse_position_expr(w.to_ident())
+            }
+            Keyword::SUBSTRING => self.parse_substring_expr(),
+            Keyword::OVERLAY => self.parse_overlay_expr(),
+            Keyword::TRIM => self.parse_trim_expr(),
+            Keyword::INTERVAL => self.parse_interval(),
+            // Treat ARRAY[1,2,3] as an array [1,2,3], otherwise try as subquery or a function call
+            Keyword::ARRAY if self.peek_token() == Token::LBracket => {
+                self.expect_token(&Token::LBracket)?;
+                self.parse_array_expr(true)
+            }
+            Keyword::ARRAY
+                if self.peek_token() == Token::LParen
+                    && !dialect_of!(self is ClickHouseDialect | DatabricksDialect) =>
+            {
+                self.expect_token(&Token::LParen)?;
+                let query = self.parse_query()?;
+                self.expect_token(&Token::RParen)?;
+                Ok(Expr::Function(Function {
+                    name: ObjectName(vec![w.to_ident()]),
+                    parameters: FunctionArguments::None,
+                    args: FunctionArguments::Subquery(query),
+                    filter: None,
+                    null_treatment: None,
+                    over: None,
+                    within_group: vec![],
+                }))
+            }
+            Keyword::NOT => self.parse_not(),
+            Keyword::MATCH if dialect_of!(self is MySqlDialect | GenericDialect) => {
+                self.parse_match_against()
+            }
+            Keyword::STRUCT if dialect_of!(self is BigQueryDialect | GenericDialect) => {
+                self.prev_token();
+                self.parse_bigquery_struct_literal()
+            }
+            Keyword::PRIOR if matches!(self.state, ParserState::ConnectBy) => {
+                let expr = self.parse_subexpr(self.dialect.prec_value(Precedence::PlusMinus))?;
+                Ok(Expr::Prior(Box::new(expr)))
+            }
+            Keyword::MAP if self.peek_token() == Token::LBrace && self.dialect.support_map_literal_syntax() => {
+                self.parse_duckdb_map_literal()
+            }
+            Keyword::DEFAULT => Ok(Expr::Default),
+            _ => Err(ParserError::BranchAbandoned)
+        }
+    }
+
+    fn parse_ident_expr(&mut self, w: &Word) -> Result<Expr, ParserError> {
```

Comment:
hmm thinking we would need to rename the function, it seems to be doing more than identifiers (and based on the name its unclear what the word `w` input argument is for)